src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -108,6 +109,7 @@ def load_arguments(self, _):
         c.argument('enable_managed_identity', action='store_true')
         c.argument('assign_identity', type=str, validator=validate_assign_identity)
         c.argument('disable_sgxquotehelper', action='store_true')
+        c.argument('auto_upgrade_channel', arg_type=get_enum_type([CONST_RAPID_UPGRADE_CHANNEL, CONST_STABLE_UPGRADE_CHANNEL, CONST_PATCH_UPGRADE_CHANNEL, CONST_NONE_UPGRADE_CHANNEL]))
```

Comment:
neatdoes arg_type get us validation here too?

Reply:
yes, I think so. I tried with some invalid strs, and it gave InvalidArgumentValueError